sdk/python/tests/test_client.py

```diff
@@ -495,516 +534,102 @@ def test_list_feature_sets(self, mocked_client, mocker):
-        driver_fs.add(Feature(name="feature_3", dtype=ValueType.INT64))
-        driver_fs.add(Entity(name="entity_id", dtype=ValueType.INT64))
-
-        # Register with Feast core
-        test_client.apply(driver_fs)
-        driver_fs = driver_fs.to_proto()
-        driver_fs.meta.status = FeatureSetStatusProto.STATUS_READY
-
-        mocker.patch.object(
-            test_client._core_service_stub,
-            "GetFeatureSet",
-            return_value=GetFeatureSetResponse(feature_set=driver_fs),
-        )
-
-        with pytest.raises(exception):
-            test_client.ingest("driver-feature-set", dataframe, timeout=1)
-
-    @pytest.mark.parametrize(
-        "dataframe,exception,test_client",
-        [
-            (dataframes.GOOD, TimeoutError, lazy_fixture("client")),
-            (dataframes.GOOD, TimeoutError, lazy_fixture("secure_client")),
-        ],
-    )
-    def test_feature_set_ingest_fail_if_pending(
-        self, dataframe, exception, test_client, mocker
-    ):
-        with pytest.raises(exception):
-            test_client.set_project("project1")
-            driver_fs = FeatureSet(
-                "driver-feature-set",
-                source=KafkaSource(brokers="kafka:9092", topic="test"),
-            )
-            driver_fs.add(Feature(name="feature_1", dtype=ValueType.FLOAT))
-            driver_fs.add(Feature(name="feature_2", dtype=ValueType.STRING))
-            driver_fs.add(Feature(name="feature_3", dtype=ValueType.INT64))
-            driver_fs.add(Entity(name="entity_id", dtype=ValueType.INT64))
-
-            # Register with Feast core
-            test_client.apply(driver_fs)
-            driver_fs = driver_fs.to_proto()
-            driver_fs.meta.status = FeatureSetStatusProto.STATUS_PENDING
-
-            mocker.patch.object(
-                test_client._core_service_stub,
-                "GetFeatureSet",
-                return_value=GetFeatureSetResponse(feature_set=driver_fs),
-            )
-
-            # Need to create a mock producer
-            with patch("feast.client.get_producer"):
-                # Ingest data into Feast
-                test_client.ingest("driver-feature-set", dataframe, timeout=1)
-
-    @pytest.mark.parametrize(
-        "dataframe,exception,test_client",
-        [
-            (dataframes.BAD_NO_DATETIME, Exception, lazy_fixture("client")),
-            (
-                dataframes.BAD_INCORRECT_DATETIME_TYPE,
-                Exception,
-                lazy_fixture("client"),
+        # Create Feature Tables
+        batch_source = DataSource(
+            type=SourceType(1).name,
+            field_mapping={
+                "ride_distance": "ride_distance",
+                "ride_duration": "ride_duration",
+            },
+            options=FileOptions(file_format="avro", file_url="data/test.avro"),
+            timestamp_column="ts_col",
+            date_partition_column="date_partition_col",
+        )
+
+        stream_source = DataSource(
```

Comment:
We should be using specific sources like `FileSource` here and not the underlying base class. Otherwise the API is too low level.

Reply:
In protos, we have the different DataSources identified via its options [here](https://github.com/feast-dev/feast/blob/master/protos/feast/core/DataSource.proto#L97). Thus, in the sdk, I only exposed DataSource, and let the different options be configurable via native classes `FileOptions`, `BigQueryOptions`, `KafkaOptions`, `KinesisOptions`.

Reply:
Is our preference here to create native classes `FileSource`, `BigQuerySource`, `KafkaSource` and `KinesisSource`? I felt that following how protos were defined would be a more standardized approach for the codebase.

Reply:
The `Source` based approach (as in the RFC) is a more abstracted view. It allows us to define a source specific constructor. So they dont have to define the type, we can define that. We can also have input arguments that are specific to each source, and we dont require them to define `FileOptions` which is basically the same as having `FileSource` at the end of the day. So its better to just have a single constructor instead of two.

Reply:
Agreed that would be cleaner. I've updated the FeatureTable class and only expose `FileSource`, `BigQuerySource`, `KafkaSource` and `KinesisSource` instead of the respective options native classes.